packages/valory/contracts/mech_mm/contract.py

```diff
@@ -0,0 +1,459 @@
+                    "name": "data",
+                    "type": "bytes",
+                },
+            ],
+            "name": "Deliver",
+            "type": "event",
+        },
+        {
+            "anonymous": False,
+            "inputs": [
+                {
+                    "indexed": True,
+                    "internalType": "address",
+                    "name": "sender",
+                    "type": "address",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "uint256",
+                    "name": "requestId",
+                    "type": "uint256",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "bytes",
+                    "name": "data",
+                    "type": "bytes",
+                },
+            ],
+            "name": "Request",
+            "type": "event",
+        },
+    ],
+    [
+        {
+            "anonymous": False,
+            "inputs": [
+                {
+                    "indexed": True,
+                    "internalType": "address",
+                    "name": "sender",
+                    "type": "address",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "uint256",
+                    "name": "requestId",
+                    "type": "uint256",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "uint256",
+                    "name": "requestIdWithNonce",
+                    "type": "uint256",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "bytes",
+                    "name": "data",
+                    "type": "bytes",
+                },
+            ],
+            "name": "Request",
+            "type": "event",
+        },
+        {
+            "anonymous": False,
+            "inputs": [
+                {
+                    "indexed": True,
+                    "internalType": "address",
+                    "name": "sender",
+                    "type": "address",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "uint256",
+                    "name": "requestId",
+                    "type": "uint256",
+                },
+                {
+                    "indexed": False,
+                    "internalType": "bytes",
+                    "name": "data",
+                    "type": "bytes",
+                },
+            ],
+            "name": "Deliver",
+            "type": "event",
+        },
+    ],
+]
+
+
+class MechMM(Contract):
+    """The Mech contract for marketplace."""
+
+    contract_id = PUBLIC_ID
+
+    @staticmethod
+    def execute_with_timeout(func: Callable, timeout: float) -> Any:
+        """Execute a function with a timeout."""
```

Comment:
https://github.com/valory-xyz/open-aea/blob/7bf2451347ed7f5168969c684e79d77657a64971/docs/api/plugins/aea_ledger_ethereum/ethereum.md?plain=1#L129

Reply:
Isn't this unrelated to this PR?